graspologic/plot/plot_matrix.py

```diff
@@ -0,0 +1,735 @@
+        sns.heatmap(
+            data,
+            cmap=cmap,
+            ax=ax,
+            center=center,
+            xticklabels=False,
+            yticklabels=False,
+            **kws,
+        )
+    elif plot_type == "scattermap":
+        gridmap(data, ax=ax, sizes=sizes, **kws)
+
+    # extra features of the graph
+    if square:
+        ax.axis("square")
+    if plot_type == "scattermap":
+        ax_pad = 0.5
+    else:
+        ax_pad = 0
+    ax.set_ylim(data.shape[0] - ax_pad, 0 - ax_pad)
+    ax.set_xlim(0 - ax_pad, data.shape[1] - ax_pad)
+    # plt.show()
+
+    # ticks, separators, colors, and spine not implemented yet
+    # make_axes_locatable allows us to create new axes like colormap
+    divider = make_axes_locatable(ax)
+
+    # draw separators
+    draw_separators(
+        ax,
+        ax_type="x",
+        meta=meta,
+        group_class=group_class,
+        plot_type=plot_type,
+        gridline_kws=gridline_kws,
+    )
+    draw_separators(
+        ax,
+        ax_type="y",
+        meta=meta,
+        group_class=group_class,
+        plot_type=plot_type,
+        gridline_kws=gridline_kws,
+    )
+    # plt.show()
+
+    # draw highlighted separators
+    if highlight_kws is None:
+        highlight_kws = dict(color="black", linestyle="-", linewidth=1)
+    if highlight is not None:
+        draw_separators(
+            ax,
+            ax_type="x",
+            meta=meta,
+            group_class=highlight,
+            plot_type=plot_type,
+            gridline_kws=highlight_kws,
+        )
+        draw_separators(
+            ax,
+            ax_type="y",
+            meta=meta,
+            group_class=highlight,
+            plot_type=plot_type,
+            gridline_kws=highlight_kws,
+        )
+
+    first_ax = True
+    # draw colors
+    if len(color_class) > 0:
+        if color_pad is None:
+            color_pad = len(color_class) * [0.5]
+        if first_ax:
+            color_pad[0] = 0
+            first_ax = False
+
+        rev_color_class = list(color_class[::-1])
+        for i, sc in enumerate(rev_color_class):
+            color_ax = divider.append_axes(
+                "top", size="3%", pad=color_pad[i], sharex=ax
+            )
+            _remove_shared_ax(color_ax)
+            draw_colors(
+                color_ax,
+                meta=meta,
+                divider=divider,
+                ax_type="x",
+                color_class=rev_color_class[i],
+                palette=palette,
+            )
+            # plt.show()
+            color_ax = divider.append_axes(
+                "left", size="3%", pad=color_pad[i], sharey=ax
+            )
+            _remove_shared_ax(color_ax)
+            draw_colors(
+                color_ax,
+                meta=meta,
+                divider=divider,
+                ax_type="y",
+                color_class=rev_color_class[i],
+                palette=palette,
+            )
+            # plt.show()
+    # plt.show()
```

Comment:
make sure you remove any "junk code" prior to getting a review

Reply:
Okay, that's my bad, I though I got rid of all of them